### PR TITLE
server: support download optimize trace file

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

support download optimize trace file

Issue Number: ref https://github.com/pingcap/tidb/issues/29661

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by ngaut** [comment]
Where is the related issue?

**Comment by Yisaer** [comment]
> Where is the related issue?

updated in description.

**Comment by JmPotato** [comment]
Do we have a manual test here🤔？

**Review by JmPotato — COMMENTED** [review]
Do we have a manual test here🤔？

**Comment by Yisaer** [comment]
> Do we have a manual test here🤔？

yes,I have manually tested it and it works well.

File: server/plan_replayer.go [hunk 1]
```diff
@@ -53,10 +53,25 @@ func (s *Server) newPlanReplayerHandler() *PlanReplayerHandler {
...
+		infoGetter:         prh.infoGetter,
+		address:            prh.address,
+		statusPort:         prh.statusPort,
+		urlPath:            fmt.Sprintf("plan_replyaer/dump/%s", name),
+		downloadedFilename: "plan_replayer",
+	}
+	handleDownloadFile(handler, w, req)
+}
+
+func handleDownloadFile(handler downloadFileHandler, w http.ResponseWriter, req *http.Request) {
```
Comment (JmPotato) [inline server/plan_replayer.go:68]: What about making this function a method of struct `downloadFileHandler`?

**Review by JmPotato — APPROVED** [review]
The rest LGTM.

**Review by rebelice — APPROVED** [review]

File: server/plan_replayer.go (hunk 1 above)
Comment (Yisaer) [reply server/plan_replayer.go:68]: not necessary I think.

**Pull request merged**